substrate/frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -783,7 +783,7 @@ fn decl_static_assertions(
 		quote! {
 			#scrate::__private::tt_call! {
 				macro = [{ #path::tt_error_token }]
-				frame_support = [{ #scrate }]
+				your_tt_return = [{ #scrate::__private::tt_return }]
```

Comment:
I recall this was one of the BIG BIG fixes in this PR when I was working on it. 

Try this for yourself: if you are familiar with the tt_call pattern a bit, instead of passing in the `your_tt_return`, pass `frame_support` as it was in the past to the caller. Then have the caller call `#frame_support::__private::tt_return` instead of plain `your_tt_return`. These two seem equivalent, but the former never worked.

Reply:
> These two seem equivalent, but the former never worked.

Yes, that's what looks at a glance... did you find out what's the reason?

Reply:
> Would be good if you also bring back some sort of example here to demonstrate that this change is actually needed

@kianenigma  Probably the best example is by using the frame-crate itself, which makes me reconsider if bringing this change here was a good idea and instead bring it in the 'main' frame umbrella PR. WDYT?

Reply:
I highly suspect that it didn't work before because the "receiver" didn't expect to see any "arguments" named `your_tt_return` and have been coded to only recognize `frame_support` as an "argument". If you take a look at `tt_default_parts.rs`, you'll see what I mean.

Reply:
If my memory is good, the former didn't work because there was another error when giving a path instead of a single ident:
Like code was something like this:
```rust
				frame_support = [{ $($frame_support:ident)::* }]
			} => {
				$($frame_support::)*__private::tt_return! {
```
but it should have been like this for path to work
```rust
				frame_support = [{ $($frame_support:ident)::* }]
			} => {
// see the semicolon is after the parentheses here
				$($frame_support)::*__private::tt_return! {
```
but when you implemented with `your_tt_return` you fixed this kind of error.

Reply:
the specific workaround for passing paths to these btw is to do an interpolation of idents segmented by `::`. For some reason that just works where `path` doesn't,

see: https://github.com/sam0x17/macro_magic/blob/main/core/src/lib.rs#L511